### PR TITLE
Update to go 1.14

### PR DESCRIPTION
**Pull request by prestonvanloon**
https://blog.golang.org/go1.14

Also replaced the cache here with LRU since tests were too flaky or failing with a sleep delay.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Blocked by https://github.com/bazelbuild/rules_go/issues/2387

**Comment by prestonvanloon** [comment]
No longer blocked, fixed in https://github.com/bazelbuild/rules_go/pull/2388.

**Comment by prestonvanloon** [comment]
Blocked again by an unsafe pointer conversion in upstream dependency. Trying to update it now...

```

fatal error: checkptr: unsafe pointer conversion
--
  |  
  | goroutine 1 [running, locked to thread]:
  | runtime.throw(0x1b2a0c1, 0x23)
  | GOROOT/src/runtime/panic.go:1112 +0x72 fp=0xc0007f7ac8 sp=0xc0007f7a98 pc=0x4762f2
  | runtime.checkptrAlignment(0xc00015d429, 0x190c260, 0x1)
  | GOROOT/src/runtime/checkptr.go:13 +0xd0 fp=0xc0007f7af8 sp=0xc0007f7ac8 pc=0x447540
  | golang.org/x/crypto/sha3.xorInUnaligned(0xc00015d340, 0xc00015d429, 0x88, 0xa8)
  | external/org_golang_x_crypto/sha3/xor_unaligned.go:13 +0x60 fp=0xc0007f7b30 sp=0xc0007f7af8 pc=0xe0b300
  | golang.org/x/crypto/sha3.(*state).permute(0xc00015d340)
  | external/org_golang_x_crypto/sha3/sha3.go:84 +0x1d3 fp=0xc0007f7b80 sp=0xc0007f7b30 pc=0xe09943
  | golang.org/x/crypto/sha3.(*state).padAndPermute(0xc00015d340, 0x29f3401)
  | external/org_golang_x_crypto/sha3/sha3.go:117 +0x347 fp=0xc0007f7c28 sp=0xc0007f7b80 pc=0xe09d07
  | golang.org/x/crypto/sha3.(*state).Read(0xc00015d340, 0xc000630640, 0x20, 0x20, 0x7f7ce631e560, 0x0, 0xc00015d180)
  | external/org_golang_x_crypto/sha3/sha3.go:163 +0x2e0 fp=0xc0007f7cb8 sp=0xc0007f7c28 pc=0xe0a7d0
  | golang.org/x/crypto/sha3.(*state).Sum(0xc00015d180, 0xc000630620, 0x0, 0x20, 0x0, 0x0, 0x0)
  | external/org_golang_x_crypto/sha3/sha3.go:190 +0x233 fp=0xc0007f7d50 sp=0xc0007f7cb8 pc=0xe0aa33
  | github.com/ethereum/go-ethereum/crypto.Keccak256Hash(0xc00063cca0, 0x1, 0x1, 0x6ef8c092e64583ff, 0xc0ad6c991be0485b, 0x21b463e3b52f6201, 0x0)
  | external/com_github_ethereum_go_ethereum/crypto/crypto.go:69 +0x1b6 fp=0xc0007f7db8 sp=0xc0007f7d50 pc=0x13c8786
  | github.com/ethereum/go-ethereum/trie.init()
  | external/com_github_ethereum_go_ethereum/trie/trie.go:34 +0xa8a fp=0xc0007f7e98 sp=0xc0007f7db8 pc=0x153048a


```

**Comment by prestonvanloon** [comment]
Seems like boltdb has this issue too. I'll put this on hold for now.

**Comment by prestonvanloon** [comment]
One solution is to upgrade to bbolt, however the issue exists there too. Blocked by https://github.com/etcd-io/bbolt/issues/187 for now.

**Comment by prestonvanloon** [comment]
One possible workaround is to disable this compiler check with `-gcflags=all=-d=checkptr=0`

https://github.com/kubernetes/kubernetes/pull/88638#issuecomment-592750265

**Comment by prestonvanloon** [comment]
Unblocked!

File: beacon-chain/operations/attestations/service.go [hunk 1]
```diff
@@ -26,11 +26,7 @@ type Config struct {
 // NewService instantiates a new attestation pool service instance that will
 // be registered into a running beacon node.
 func NewService(ctx context.Context, cfg *Config) (*Service, error) {
-	cache, err := ristretto.NewCache(&ristretto.Config{
-		NumCounters: forkChoiceProcessedRootsSize,
-		MaxCost:     forkChoiceProcessedRootsSize,
-		BufferItems: 64,
-	})
+	cache, err := lru.New(forkChoiceProcessedRootsSize)
```
Comment (terencechain) [inline beacon-chain/operations/attestations/service.go]: I think it's ok to bundle this change in the same PR, perhaps just mention it in the description?

Comment (prestonvanloon) [reply beacon-chain/operations/attestations/service.go]: oh yes good idea

**Pull request merged**